### PR TITLE
EO-852: Added RV price modal for RV page

### PR DESCRIPTION
**Pull request by PatrickSison**
### What Changed
 - Added button to link to price modal
 - Added price modal

### How To Test
 - Navigate to `/recipient-validation`
 - Check that button on top right says `See Pricing`
 - Check that it opens modal

### To Do
- [ ] Address feedback

### CODE REVIEW

File: src/pages/recipientValidation/components/RecipientValidationPriceModal.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+const RecipientValidationPriceModal = ({ onClose }) => (
+  <Panel
+    className={styles.modalContainer}
+    accent
+  >
+    <div style={{ float: 'right' }}><Button onClick={() => onClose()} flat><Close /></Button></div>
+    <div className={styles.bodyContainer}>
+      <h3>How was this calculated?</h3>
+      <Table className={styles.table}>
```
Comment (kchu1337) [inline src/pages/recipientValidation/components/RecipientValidationPriceModal.js]: The same table exists within the RV disabled page. If possible, let's make it into a separate component and just that component in both cases.

File: src/pages/recipientValidation/RecipientValidationPage.module.scss [hunk 2]
```diff
@@ -3,3 +3,10 @@
 .TabPadding {
   padding: 0 spacing();
 }
+
+.LeadText {
+  font-size: small;
+  max-width: 60%;
```
Comment (kchu1337) [inline src/pages/recipientValidation/RecipientValidationPage.module.scss]: The width looks a little larger in the sketch.

File: src/pages/recipientValidation/RecipientValidationPage.js [hunk 3]
```diff
@@ -58,6 +69,9 @@ export class RecipientValidationPage extends Component {
           {this.renderTabContent(selectedTab)}
         </Panel>
         {selectedTab === 0 && <ListResults/>}
+        <Modal open={showPriceModal} onClose={() => this.handleModal()}>
```
Comment (kchu1337) [inline src/pages/recipientValidation/RecipientValidationPage.js]: I would add `false` in the parameter so it's obvious that it's closing the modal. If you decide to keep the current handleModal functionality.

File: src/pages/recipientValidation/RecipientValidationPage.js [hunk 4]
```diff
@@ -43,12 +45,21 @@ export class RecipientValidationPage extends Component {
     }
   }
 
+  handleModal = (showPriceModal = false) => this.setState({ showPriceModal });
```
Comment (kchu1337) [inline src/pages/recipientValidation/RecipientValidationPage.js]: In other pages with modals the handle modal function does 
`this.setState({showPriceModal: !this.state.showPriceModal})`. Or has separate open/close functions.

File: src/pages/recipientValidation/components/RecipientValidationPriceModal.module.scss [hunk 5]
```diff
@@ -0,0 +1,46 @@
+@import '~@sparkpost/matchbox/src/styles/config.scss';
```
Comment (kchu1337) [inline src/pages/recipientValidation/components/RecipientValidationPriceModal.module.scss]: I think you have some extra classes that is not currently used?

File: src/pages/recipientValidation/RecipientValidationPage.js [hunk 6]
```diff
@@ -43,12 +45,21 @@ export class RecipientValidationPage extends Component {
...
   renderRecipientValidation = () => {
-    const { selectedTab } = this.state;
+    const { selectedTab, showPriceModal } = this.state;
 
     return (
       <Page
-        title='Recipient Validation'>
+        title='Recipient Validation'
+        primaryArea={<Button size='large' onClick={() => this.handleModal(true)}>See Pricing</Button>}>
+        <p className={styles.LeadText}>
```
Comment (kchu1337) [inline src/pages/recipientValidation/RecipientValidationPage.js]: There is some overlap in the RV disabled page. Let's do something about that. Same for the table in your pricing modal.

File: src/pages/recipientValidation/components/RecipientValidationPriceModal.module.scss [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  margin: 0 auto;
+}
+
+.bodyContainer {
+  padding: 2rem 3rem;
+}
+
+.tableHeader {
+  color: color(orange);
+  font-weight: 100;
```
Comment (kchu1337) [inline src/pages/recipientValidation/components/RecipientValidationPriceModal.module.scss]: The sketch text looks a little bolder?

File: src/pages/recipientValidation/RecipientValidationPage.module.scss (hunk 2 above)
Comment (PatrickSison) [reply src/pages/recipientValidation/RecipientValidationPage.module.scss]: I'm using the same styling as the new templates page. Lynn gave the OK for this one

File: src/pages/recipientValidation/components/RecipientValidationPriceTable.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Table } from '@sparkpost/matchbox';
+import { RECIPIENT_VALIDATION_TIERS } from 'src/constants';
+import { formatFullNumber } from 'src/helpers/units';
+import styles from './RecipientValidationPriceTable.module.scss';
+
+export const RecipientValidationPriceTable = ({ cellProps, col1Props, col2Props }) => (
+  <Table className={styles.table}>
```
Comment (kchu1337) [inline src/pages/recipientValidation/components/RecipientValidationPriceTable.js]: I don't think this css class exists anymore.

File: src/pages/recipientValidation/components/RVDisabledPage.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import { Component } from 'react';
-import { Page, Panel, Table, Button } from '@sparkpost/matchbox';
+import { Page, Panel, Button } from '@sparkpost/matchbox';
 import React from 'react';
 import { Link } from 'react-router-dom';
 import styles from './RVDisabledPage.module.scss';
```
Comment (kchu1337) [inline src/pages/recipientValidation/components/RVDisabledPage.js]: You can delete some of the CSS classes that are no longer used anymore in this file.

**Pull request closed without merging**